File: BE/health-service/src/main/java/com/s206/health/step/dto/response/StepWeeklyResponse.java
package com.s206.health.step.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class StepWeeklyResponse {
    private List<StepDailyData> weeklySteps;
    private LocalDate startDate;
    private LocalDate endDate;
    private Integer totalSteps;
    private Integer avgSteps;

    @Getter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class StepDailyData {
        private LocalDate date;
        private Integer stepCount;
    }
}

File: FE/mobile/src/main/java/com/example/diaviseo/network/exercise/dto/res/StepWeeklyResponse.kt
package com.example.diaviseo.network.exercise.dto.res

data class StepWeeklyResponse(
    val weeklySteps: List<StepDailyData>,
    val startDate: String,
    val endDate: String,
    val totalSteps: Int,
    val avgSteps: Int
) {
    data class StepDailyData(
        val date: String,
        val stepCount: Int
    )
}